testing/src/main/kotlin/com/airbnb/mvrx/test/MvRxTestRule.kt

```diff
@@ -0,0 +1,83 @@
+package com.airbnb.mvrx.test
+
+import android.support.annotation.NonNull
+import com.airbnb.mvrx.MvRxTestOverridesProxy
+import io.reactivex.Scheduler
+import io.reactivex.android.plugins.RxAndroidPlugins
+import io.reactivex.disposables.Disposable
+import io.reactivex.exceptions.CompositeException
+import io.reactivex.internal.schedulers.ExecutorScheduler
+import io.reactivex.plugins.RxJavaPlugins
+import io.reactivex.schedulers.Schedulers
+import org.junit.rules.TestRule
+import org.junit.runner.Description
+import org.junit.runners.model.Statement
+import java.util.concurrent.Executor
+import java.util.concurrent.TimeUnit
+
+enum class DebugMode(internal val value: Boolean?) {
+    Debug(true),
+    NotDebug(false),
+    Unset(null)
+}
+
+class MvRxTestRule(
+        /**
+         * Forces MvRx to be in debug mode or not.
+         */
+        private val debugMode: DebugMode = DebugMode.NotDebug,
+        /**
+         * Sets up all Rx schedulers to use an immediate scheduler. This will cause all MvRx
+         * operations including setState reducers to run synchronously so you can test them.
+         */
+        private val setRxImmediateSchedulers: Boolean = true
+) : TestRule {
+    private val immediateScheduler = object : Scheduler() {
+        // this prevents StackOverflowErrors when scheduling with a delay
+        override fun scheduleDirect(@NonNull run: Runnable, delay: Long, @NonNull unit: TimeUnit): Disposable = super.scheduleDirect(run, 0, unit)
+
+        override fun createWorker(): Scheduler.Worker = ExecutorScheduler.ExecutorWorker(Executor { it.run() })
+    }
+
+    private var defaultExceptionHandler: Thread.UncaughtExceptionHandler? = null
+
+    override fun apply(base: Statement, description: Description): Statement {
+        return object : Statement() {
+            override fun evaluate() {
+                RxAndroidPlugins.reset()
+                RxAndroidPlugins.setInitMainThreadSchedulerHandler { Schedulers.trampoline() }
```

Comment:
@gpeal Not a super strong reason. I chose trampoline because the main thread scheduler gets stored as a static variable in AndroidSchedulers, so I thought it was better to store an RxJava scheduler than our own. But, Trampoline and immediate are very similar. Do you recall if there was a reason why we were using our own immediate scheduler? If not, we should consider just switching to all trampolines. I verified the tests still pass with using trampoline everywhere.